examples/code/integrations/twitch/custom_tool.py

```diff
@@ -0,0 +1,50 @@
+from typing import Annotated, Optional
+
+import httpx
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Twitch
+
+
+@tool(
+    requires_auth=Twitch(
+        scopes=["channel:manage:polls"],
+    )
+)
+async def create_poll(
+    context: ToolContext,
+    broadcaster_id: Annotated[
+        str,
+        "The ID of the broadcaster to create the poll for.",
+    ],
+    title: Annotated[
+        str,
+        "The title of the poll.",
+    ],
+    choices: Annotated[
+        list[str],
+        "The choices of the poll.",
+    ],
+    duration: Annotated[
+        int,
+        "The duration of the poll in seconds.",
+    ],
+) -> Annotated[dict, "The poll that was created"]:
+    """Create a poll for a Twitch channel."""
+    url = "https://api.twitch.tv/helix/polls"
+    headers = {
+        "Authorization": f"Bearer {context.authorization.token}",
+        "Client-Id": "your_client_id",
```

Comment:
I don't think I've ever seen the need for the client ID of the app to be in the header. Does the `ToolContext` contain this information?

Reply:
Yes, this is very strange.
`ToolContext` does not contain the info, so it would have to be hardcoded for now. Filed: https://app.clickup.com/t/86b4a5bvj